homeassistant/components/camera/__init__.py

```diff
@@ -277,29 +279,39 @@ def handle_async_mjpeg_stream(self, request):
 
         try:
             while True:
-                img_bytes = yield from self.async_camera_image()
+                img_bytes = await self.async_camera_image()
                 if not img_bytes:
                     break
 
                 if img_bytes and img_bytes != last_image:
-                    yield from write(img_bytes)
+                    await write_to_mjpeg_stream(img_bytes)
 
                     # Chrome seems to always ignore first picture,
                     # print it twice.
                     if last_image is None:
-                        yield from write(img_bytes)
+                        await write_to_mjpeg_stream(img_bytes)
 
                     last_image = img_bytes
 
-                yield from asyncio.sleep(.5)
+                await asyncio.sleep(interval)
```

Comment:
This interval does not work as advertised. It's the interval that we wait between finishing writing a frame and before we start fetching the next frame. What people assume when they see the word interval is that it is the time between the start of writing two frames. (note that currently it is not configurable and thus everything worked as advertised)

Reply:
I explicitly picked interval, and not ‘refresh rate’ or ‘FPS’ because I was already aware of this.